lib/middlewares/responseGenerator.js

```diff
@@ -105,6 +105,30 @@ function blackButtonResponse(parsedBody, req) {
     return payload;
 }
 
+function blackButtonError(parsedBody, req) {
+    var payload,
+        details,
+        code,
+        errorType;
+
+    if (req.cbError) {
+        if (req.cbError.code && req.cbError.code !== 200) {
+            code = req.cbError.code;
+            errorType = 0;
+            details = '';
+        } else {
+            code = req.cbError.details.code || 'UNK';
```

Comment:
Just wanted to cross-check with you if in case `req.cbError` is truthy and `(req.cbError.code && req.cbError.code !== 200)` is falsy, 'req.cbError.details' always exists as an Object... Otherwise, this line and the next one (but first this line obviously :D) could throw an error :)